src/datepicker/datepicker-input.ts

```diff
@@ -292,6 +294,7 @@ export class NgbInputDatepicker implements OnChanges,
           }
         });
     datepickerInstance.startDate = this.startDate || this._model;
+    datepickerInstance.setDisabledState(this._disabled);
```

Comment:
This is not an input, so I would do it in the `open()` method after the focus handling. Plus you won't need to change the service